File: src/main/java/arrays/TwoSum.java
package arrays;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class TwoSum{
  public static void main(String[] args){
    TwoSum s = new TwoSum();
    System.out.println(Arrays.toString(s.twoSum(new int[]{2, 7, 11, 13}, 9)));
  }

  public int[] twoSum(int[] nums, int target) {
    Map<Integer, Integer> diffMap = new HashMap<>();
    int diff = -1;
    for(int i = 0; i < nums.length; ++i){
      int curNum = nums[i];
      diff = target - curNum;
      if(diffMap.containsKey(diff)){
        return new int[]{diffMap.get(diff), i};
      } else {
        diffMap.put(curNum, i);
      }
    }
    return new int[]{-1, -1};
  }
}

File: src/TwoSum.kt
fun main(){
    println(twoSum(intArrayOf(2, 7, 11, 21), 18).joinToString())
}

fun twoSum(nums: IntArray, target: Int): IntArray {
    val diffMap = HashMap<Int, Int>(nums.size);
    for(i in 0 until nums.size){
        val curNum = nums[i]
        val diff = target - curNum
        if(diffMap.containsKey(diff)){
            return intArrayOf(diffMap[diff]?:0, i)
        }
        diffMap[curNum] = i
    }
    return intArrayOf(-1, -1)
}